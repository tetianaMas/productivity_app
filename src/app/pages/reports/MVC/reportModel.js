import { eventBus } from '../../../services/eventBus';
import { Utils } from '../../../helpers/utils';

/**
 * @description controls reports data
 * @exports ReportModel
 * @class ReportModel
 */
export class ReportModel {
  /**
   * @description Creates an instance of ReportModel
   * @param {object} dataService an instance of dataService
   * @param {object} Observer instance of ReportObserver
   * @memberof ReportModel
   */
  constructor(dataService, Observer) {
    this.dataService = dataService;
    this.dataTabs = {
      additionalClasses: 'tab--center js-tab-top',
      firstVal: 'day',
      firstValAttr: 'data-reports-period="day"',
      secondVal: 'week',
      secondValAttr: 'data-reports-period="week"',
      thirdVal: 'month',
      thirdValAttr: 'data-reports-period="month"',
      period: true,
      bottomTabs: {
        additionalClasses: 'tab--center tab--bottom js-tab-bottom',
        firstVal: 'pomodoros',
        firstValAttr: 'data-reports-category="pomodoros"',
        secondVal: 'tasks',
        secondValAttr: 'data-reports-category="tasks"',
        secondValActive: true,
      },
    };
    this.periods = {
      day: 1,
      week: 7,
      month: 30,
    };
    this.categories = {
      pomodoros: this.createDataPomodoros.bind(this),
      tasks: this.createDataTasks.bind(this),
    };

    this.tasksByCurrentPeriod = null;

    this.renderPageContainerEvent = new Observer();
    this.dataReadyEvent = new Observer();
  }

  /**
   * @description renders page
   * @param {string} category report category
   * @param {string} period report period
   * @memberof ReportModel
   */
  async createPage(category, period) {
    try {
      this.renderPageContainerEvent.notify(this.dataTabs);
      await this.getAndSaveData('tasks');

      const currentPeriod = this.periods[period];

      if (
        typeof category === 'string' &&
        typeof period === 'string' &&
        currentPeriod
      ) {
        this.createReport(category, currentPeriod, period);
      }
    } catch (err) {
      console.error(err);
    }
  }

  /**
   * @description receives and saves data for reports
   * @param {string} key key for receiving data from firebase
   * @memberof ReportModel
   */
  async getAndSaveData(key) {
    try {
      this.tasks = await this.dataService.receiveData(key);
    } catch (err) {
      console.error(err.message);
    }
  }

  /**
   * @description creates report from completed tasks
   * @param {string} category report category
   * @param {number} periodMeasure report category
   * @param {string} period report period
   * @memberof ReportModel
   */
  createReport(category, periodMeasure, period) {
    const completedTasks = this.tasks.filter(
      task => task.status === 'COMPLETED'
    );

    if (
      Array.isArray(completedTasks) &&
      completedTasks.length &&
      periodMeasure &&
      typeof periodMeasure === 'number' &&
      period
    ) {
      const tasksByCurrentPeriod = this.sortReportsByPeriod(
        completedTasks,
        periodMeasure
      );
      this.createReportsData(category, tasksByCurrentPeriod, period);
    }
  }

  /**
   * @description sorts all completed tasks by date
   * @param {object} completedTasks all completed tasks
   * @param {number} daysInReport number of days in current report
   * @return {object} sorted tasks
   * @memberof ReportModel
   */
  sortReportsByPeriod(completedTasks, daysInReport) {
    if (
      !Array.isArray(completedTasks) ||
      !completedTasks.length ||
      !daysInReport ||
      typeof daysInReport !== 'number'
    ) {
      return;
    }
    let tasksByCurrentPeriod;

    if (daysInReport > 1) {
      const MILLISECONDS_IN_DAY = 86400000;
      const fullTimeReport = daysInReport * MILLISECONDS_IN_DAY;
      const currentDateMilliseconds = new Date().getTime();
      const lastDayInPeriod = currentDateMilliseconds - fullTimeReport;

      tasksByCurrentPeriod = completedTasks.filter(
        task =>
          Utils.getNumberDate(task.completeDate).getTime() > lastDayInPeriod
      );
    } else {
      const currentDateString = new Date();

      tasksByCurrentPeriod = completedTasks.filter(
        task =>
          Utils.getNumberDate(task.completeDate).getDate() ===
          currentDateString.getDate()
      );
    }

    return tasksByCurrentPeriod.sort(
      (a, b) =>
        Utils.getNumberDate(a.completeDate).getTime() -
        Utils.getNumberDate(b.completeDate).getTime()
    );
  }

  /**
   * @description creates data for render, emits event for rendering
   * @param {string} category current category
   * @param {object} tasks sorted completed tasks
   * @param {number} period current period
   * @memberof ReportModel
   */
  createReportsData(category, tasks, period) {
    const dataForRender = this.categories[category](tasks);
    eventBus.post('create-report', [dataForRender, period]);
  }

  /**
   * @description creates report by pomodoros
   * @param {object} tasks sorted completed tasks
   * @return {object} object for rendering
   * @memberof ReportModel
   */
  createDataPomodoros(tasks) {
    if (!Array.isArray(tasks) || !tasks.length) {
      return;
    }
    const tasksToRender = {};

    tasks.forEach(function (task) {
      if (!tasksToRender[task.completeDate]) {
        const dataByOneDay = {
          date: task.completeDate,
          urgent: 0,
          high: 0,
          middle: 0,
          low: 0,
          failed: 0,
        };
        if (task.completedCount) {
          dataByOneDay[task.priority] += task.completedCount.length;
        }
        if (task.failedPomodoros) {
          dataByOneDay.failed += task.failedPomodoros.length;
        }
        tasksToRender[task.completeDate] = dataByOneDay;
      } else {
        if (task.completedCount) {
          tasksToRender[task.completeDate][task.priority] +=
            task.completedCount.length;
        }
        if (task.failedPomodoros) {
          tasksToRender[task.completeDate].failed +=
            task.failedPomodoros.length;
        }
      }
    });

    return Object.keys(tasksToRender).map(item => tasksToRender[item]);
  }

  /**
   * @description creates report by tasks
   * @param {object} tasks sorted completed tasks
   * @return {object} object for rendering
   * @memberof ReportModel
   */
  createDataTasks(tasks) {
    if (!Array.isArray(tasks) || !tasks.length) {
      return;
    }
    const tasksToRender = {};
    const isSuccessfullTask = this.isSuccessfullTask.bind(this);

    tasks.forEach(function (task) {
      if (isSuccessfullTask(task)) {
        if (!tasksToRender[task.completeDate]) {
          const dataByOneDay = {
            date: task.completeDate,
            urgent: 0,
            high: 0,
            middle: 0,
            low: 0,
            failed: 0,
          };
          dataByOneDay[task.priority]++;
          tasksToRender[task.completeDate] = dataByOneDay;
        } else {
          tasksToRender[task.completeDate][task.priority]++;
        }
      } else {
        if (!tasksToRender[task.completeDate]) {
          const dataByOneDay = {
            date: task.completeDate,
            urgent: 0,
            high: 0,
            middle: 0,
            low: 0,
            failed: 0,
          };
          dataByOneDay.failed++;
          tasksToRender[task.completeDate] = dataByOneDay;
        } else {
          tasksToRender[task.completeDate].failed++;
        }
      }
    });

    return Object.keys(tasksToRender).map(item => tasksToRender[item]);
  }

  /**
   * @description checks successful task
   * @param {object} task task to check
   * @return {boolean} true if successful
   * @memberof ReportModel
   */
  isSuccessfullTask(task) {
    if (!task || typeof task !== 'object') {
      return;
    }
    const failedPomodoros = task.failedPomodoros || [];
    const completedPomodoros = task.completedCount || [];

    return failedPomodoros.length <= completedPomodoros.length;
  }
}
