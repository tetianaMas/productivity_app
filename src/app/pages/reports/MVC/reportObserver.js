/**
 * @description manages custom events for report component
 * @exports ReportObserver
 * @class ReportObserver
 */
export class ReportObserver {
  /**
   * @description Creates an instance of ReportObserver
   * @memberof ReportObserver
   */
  constructor() {
    this.observers = [];
  }

  /**
   * @description adds new callback to the observers
   * @param {object} handler callback to hangle event
   * @memberof ReportObserver
   */
  subscribe(handler) {
    this.observers.push(handler);
  }

  /**
   * @description emits all handlers
   * @param {object} data arguments for the handler
   * @memberof ReportObserver
   */
  notify(data) {
    this.observers.forEach(handler => handler(data));
  }
}
