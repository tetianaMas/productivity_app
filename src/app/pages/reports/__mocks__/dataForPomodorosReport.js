import { Utils } from '../../../helpers/utils';

export const dataForPomodorosReport = [
  {
    date: Utils.getCurrentDate(),
    failed: 7,
    high: 1,
    low: 0,
    middle: 0,
    urgent: 3,
  },
];
