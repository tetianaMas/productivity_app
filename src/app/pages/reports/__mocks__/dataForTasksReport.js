import { Utils } from '../../../helpers/utils';

export const dataForTasksReport = [
  {
    date: Utils.getCurrentDate(),
    failed: 1,
    high: 0,
    low: 0,
    middle: 0,
    urgent: 1,
  },
];
