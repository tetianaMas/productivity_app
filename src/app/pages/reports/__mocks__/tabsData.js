export const tabsData = {
  additionalClasses: 'tab--center js-tab-top',
  firstVal: 'day',
  firstValAttr: 'data-reports-period="day"',
  secondVal: 'week',
  secondValAttr: 'data-reports-period="week"',
  thirdVal: 'month',
  thirdValAttr: 'data-reports-period="month"',
  period: true,
  bottomTabs: {
    additionalClasses: 'tab--center tab--bottom js-tab-bottom',
    firstVal: 'pomodoros',
    firstValAttr: 'data-reports-category="pomodoros"',
    secondVal: 'tasks',
    secondValAttr: 'data-reports-category="tasks"',
    secondValActive: true,
  },
};
