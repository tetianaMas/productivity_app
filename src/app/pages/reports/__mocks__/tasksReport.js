import { Utils } from '../../../helpers/utils';

export const tasks = [
  {
    categoryId: 'hobby',
    completeDate: Utils.getCurrentDate(),
    completedCount: [0],
    createDate: { day: 17, fullDeadline: 1615978347385, month: 'March' },
    deadline: 'April 10, 2021',
    deadlineDate: { day: '', fullDeadline: 1616716800000, month: 'today' },
    description: 'Wow',
    estimation: '5',
    failedPomodoros: [0, 1, 2, 3, 4],
    id: '00807824925639886039',
    priority: 'high',
    status: 'COMPLETED',
    title: 'Wow',
  },
  {
    categoryId: 'hobby',
    completeDate: Utils.getCurrentDate(),
    completedCount: [2, 3, 4],
    createDate: { day: 2, fullDeadline: 1617314051232, month: 'April' },
    deadline: 'April 15, 2021',
    deadlineDate: { day: 15, fullDeadline: 1618434000000, month: 'today' },
    description: 'Cool',
    estimation: '3',
    failedPomodoros: [0, 1],
    id: '08835933769418409683',
    priority: 'urgent',
    status: 'COMPLETED',
    title: 'Cool',
  },
];
