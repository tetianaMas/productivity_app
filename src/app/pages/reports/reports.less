@import (reference) '~@less-helpers-module/_mixins.less';
@import (reference) '~@less-helpers-module/_variables.less';

.main {
  &__reports {
    margin: 213/@vh 0 191/@vh;
    .flex-center();

    &__link {
      .navigation-arrow();
      .btn-link();

      &--left {
        .navigation-arrow();
        left: 0;
      }
    }

    &__container {
      position: relative;
      max-width: 100%;
      max-height: 100%;
      color: @white-color;
      .tooltip();

      table {
        border: 1/@rem solid @white-color;
        border-collapse: collapse;

        th {
          #fonts>.heading2(28/@rem, 28/@rem, 700, capitalize);
          border: 1/@rem solid @white-color;
          padding: 20/@rem;
        }

        td {
          #fonts>.heading2(24/@rem, 24/@rem, 300, capitalize);
          border: 1/@rem solid @white-color;
          text-align: center;
          padding: 20/@rem;
        }
      }


    }
  }

  .tabs {

    &--bottom {
      margin: 0;
      height: 16/@rem;

      .tab {
        margin: 0;

        &__btn::after {
          bottom: -4/@rem;
        }
      }
    }
  }

  .tabs.tabs--bottom {
    .tab {
      margin: 0;
      height: 16/@rem;

      &__btn::after {
        bottom: -4/@rem;
      }
    }
  }
}

.reports-tablet();
.reports-desktop();