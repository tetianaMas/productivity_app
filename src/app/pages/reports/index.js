import './reports.less';
import { dataService } from '../../services/dataService/dataService';
import { ReportController } from './MVC/reportController';
import { ReportModel } from './MVC/reportModel';
import { ReportObserver } from './MVC/reportObserver';
import { ReportView } from './MVC/reportView';

const reportModel = new ReportModel(dataService, ReportObserver);
const reportView = new ReportView(reportModel);
/* eslint-disable no-unused-vars */
const reportController = new ReportController(reportModel, reportView);
/* eslint-disable no-unused-vars */
