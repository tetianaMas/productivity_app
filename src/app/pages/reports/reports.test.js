/* eslint-disable no-undef */
import reports from './templates/reports.hbs';
import { ReportModel } from './MVC/reportModel';
import { dataService } from '../../services/dataService/dataService';
import { ReportObserver } from './MVC/reportObserver';
import { tasks } from './__mocks__/tasksReport';
import { tabsData } from './__mocks__/tabsData';
import { dataForPomodorosReport } from './__mocks__/dataForPomodorosReport';
import { dataForTasksReport } from './__mocks__/dataForTasksReport';

jest.mock('./MVC/reportView.js');
jest.mock('../../services/dataService/dataService');

const reportModel = new ReportModel(dataService, ReportObserver);

describe('check report rendering', () => {
  test('renders correctly report', () => {
    const tree = JSON.stringify(reports(tabsData));
    expect(tree).toMatchSnapshot();
  });
});

describe('check report rendering', () => {
  test('renders correctly report', () => {
    const tree = JSON.stringify(reports(tabsData));
    expect(tree).toMatchSnapshot();
  });
});

describe('createReport', () => {
  beforeEach(() => {
    reportModel.tasks = tasks;
    spyOn(reportModel, 'sortReportsByPeriod');
    spyOn(reportModel, 'createReportsData');
  });

  test('shold start creating reports', () => {
    reportModel.createReport('tasks', 1, 'day');
    expect(reportModel.sortReportsByPeriod).toBeCalled();
    expect(reportModel.createReportsData).toBeCalled();
  });

  test('shold not start creating reports', () => {
    reportModel.createReport('tasks');
    expect(reportModel.sortReportsByPeriod).not.toBeCalled();
    expect(reportModel.createReportsData).not.toBeCalled();
  });
});

describe('should sort tasks', () => {
  test('shold sort reports for 1 day', () => {
    const result = reportModel.sortReportsByPeriod(tasks, 1);
    expect(result).toEqual(tasks);
  });

  test('shold sort reports for 7 weeks', () => {
    const result = reportModel.sortReportsByPeriod(tasks, 7);
    expect(result).toEqual(tasks);
  });

  test('shold return falsy result', () => {
    const result = reportModel.sortReportsByPeriod(true, 7);
    expect(result).toBeFalsy();
  });

  test('shold return falsy result', () => {
    const result = reportModel.sortReportsByPeriod(true, 0);
    expect(result).toBeFalsy();
  });
});

describe('should create data for pomodoros', () => {
  test('shold return data for pomodoros report', () => {
    const result = reportModel.createDataPomodoros(tasks);
    expect(result).toEqual(dataForPomodorosReport);
  });

  test('shold return falsy result', () => {
    const result = reportModel.createDataPomodoros();
    expect(result).toBeFalsy();
  });

  test('shold return falsy result', () => {
    const result = reportModel.createDataPomodoros([]);
    expect(result).toBeFalsy();
  });

  test('shold return falsy result', () => {
    const result = reportModel.createDataPomodoros(67);
    expect(result).toBeFalsy();
  });
});

describe('should create data for tasks report', () => {
  test('shold return data for tasks report', () => {
    const result = reportModel.createDataTasks(tasks);
    expect(result).toEqual(dataForTasksReport);
  });

  test('shold return falsy result', () => {
    const result = reportModel.createDataTasks([]);
    expect(result).toBeFalsy();
  });

  test('shold return falsy result', () => {
    const result = reportModel.createDataTasks();
    expect(result).toBeFalsy();
  });

  test('shold return falsy result', () => {
    const result = reportModel.createDataTasks(6);
    expect(result).toBeFalsy();
  });
});

describe('should check if task successfull', () => {
  test('shold return truthy result', () => {
    const result = reportModel.isSuccessfullTask(tasks[1]);
    expect(result).toBeTruthy();
  });

  test('shold return falsy result', () => {
    const result = reportModel.isSuccessfullTask(tasks[0]);
    expect(result).toBeFalsy();
  });

  test('shold return falsy result', () => {
    const result = reportModel.isSuccessfullTask();
    expect(result).toBeFalsy();
  });

  test('shold return falsy result', () => {
    const result = reportModel.isSuccessfullTask(6);
    expect(result).toBeFalsy();
  });
});
