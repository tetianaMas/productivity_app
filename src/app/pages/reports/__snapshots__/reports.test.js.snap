// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check report rendering renders correctly report 1`] = `"\\" <header class=\\\\\\"header js-header-main\\\\\\">\\\\r\\\\n\\\\r\\\\n <h1 class=\\\\\\"header__title\\\\\\">\\\\r\\\\n    daily task list\\\\r\\\\n  </h1>\\\\r\\\\n  <div class=\\\\\\"header__logo d-none\\\\\\">\\\\r\\\\n    <a href=\\\\\\"#\\\\\\">\\\\r\\\\n      <img src=\\\\\\"../images/Logo.svg\\\\\\" alt=\\\\\\"Logo\\\\\\" />\\\\r\\\\n    </a>\\\\r\\\\n  </div>\\\\r\\\\n  <nav class=\\\\\\"navigation\\\\\\" aria-label=\\\\\\"Main navigation\\\\\\">\\\\r\\\\n    <ul class=\\\\\\"navigation__list\\\\\\">\\\\r\\\\n      <li class=\\\\\\"navigation__item\\\\\\">\\\\r\\\\n        <button class=\\\\\\"navigation__btn js-header-nav-btn\\\\\\" data-link=\\\\\\"task-list\\\\\\" title=\\\\\\"Go to Tasks\\\\\\">\\\\r\\\\n          <span class=\\\\\\"icon-list \\\\\\"></span>\\\\r\\\\n        </button>\\\\r\\\\n      </li>\\\\r\\\\n      <li class=\\\\\\"navigation__item\\\\\\">\\\\r\\\\n        <button class=\\\\\\"navigation__btn js-header-nav-btn\\\\\\" data-link=\\\\\\"reports/day/tasks\\\\\\" title=\\\\\\"Go to Reports\\\\\\">\\\\r\\\\n          <span class=\\\\\\"icon-statistics \\\\\\"></span>\\\\r\\\\n        </button>\\\\r\\\\n      </li>\\\\r\\\\n      <li class=\\\\\\"navigation__item\\\\\\">\\\\r\\\\n        <button class=\\\\\\"navigation__btn js-header-nav-btn\\\\\\" data-link=\\\\\\"settings/pomodoros\\\\\\" title=\\\\\\"Go to Settings\\\\\\">\\\\r\\\\n          <span class=\\\\\\"icon-settings \\\\\\"></span>\\\\r\\\\n        </button>\\\\r\\\\n      </li>\\\\r\\\\n    </ul>\\\\r\\\\n  </nav>\\\\r\\\\n</header><main id=\\\\\\"main\\\\\\" class=\\\\\\"main main--static\\\\\\">\\\\r\\\\n  <div class=\\\\\\"tabs js-nav\\\\\\">\\\\r\\\\n  \\\\r\\\\n    <div class=\\\\\\"tab tab--right tab--center js-tab-top\\\\\\">\\\\r\\\\n        <button class=\\\\\\"tab__btn js-tablink first-main-tab \\\\\\" data-reports-period=\\\\\\"day\\\\\\">day</button>\\\\r\\\\n        <button class=\\\\\\"tab__btn js-tablink second-main-tab \\\\\\" data-reports-period=\\\\\\"week\\\\\\">week</button>\\\\r\\\\n        <button class=\\\\\\"tab__btn js-tablink \\\\\\" data-reports-period=\\\\\\"month\\\\\\">month</button>\\\\r\\\\n    </div>\\\\r\\\\n  </div>\\\\r\\\\n  \\\\r\\\\n  <div class=\\\\\\"main__reports\\\\\\">\\\\r\\\\n    <a href=\\\\\\"/#/task-list\\\\\\" class=\\\\\\"main__reports__link main__reports__link--left icon-arrow-left\\\\\\" title=\\\\\\"Go to Global List\\\\\\"></a>\\\\r\\\\n    <div class=\\\\\\"main__reports__container js-reports-container\\\\\\" id=\\\\\\"reports-container\\\\\\">\\\\r\\\\n    </div>\\\\r\\\\n  </div>\\\\r\\\\n  <div class=\\\\\\"tabs js-nav\\\\\\">\\\\r\\\\n  \\\\r\\\\n    <div class=\\\\\\"tab tab--right tab--center tab--bottom js-tab-bottom\\\\\\">\\\\r\\\\n        <button class=\\\\\\"tab__btn js-tablink first-main-tab \\\\\\" data-reports-category=\\\\\\"pomodoros\\\\\\">pomodoros</button>\\\\r\\\\n        <button class=\\\\\\"tab__btn js-tablink second-main-tab \\\\\\" data-reports-category=\\\\\\"tasks\\\\\\">tasks</button>\\\\r\\\\n    </div>\\\\r\\\\n  </div>\\\\r\\\\n</main>\\""`;

exports[`check report rendering renders correctly report 2`] = `"\\" <header class=\\\\\\"header js-header-main\\\\\\">\\\\r\\\\n\\\\r\\\\n <h1 class=\\\\\\"header__title\\\\\\">\\\\r\\\\n    daily task list\\\\r\\\\n  </h1>\\\\r\\\\n  <div class=\\\\\\"header__logo d-none\\\\\\">\\\\r\\\\n    <a href=\\\\\\"#\\\\\\">\\\\r\\\\n      <img src=\\\\\\"../images/Logo.svg\\\\\\" alt=\\\\\\"Logo\\\\\\" />\\\\r\\\\n    </a>\\\\r\\\\n  </div>\\\\r\\\\n  <nav class=\\\\\\"navigation\\\\\\" aria-label=\\\\\\"Main navigation\\\\\\">\\\\r\\\\n    <ul class=\\\\\\"navigation__list\\\\\\">\\\\r\\\\n      <li class=\\\\\\"navigation__item\\\\\\">\\\\r\\\\n        <button class=\\\\\\"navigation__btn js-header-nav-btn\\\\\\" data-link=\\\\\\"task-list\\\\\\" title=\\\\\\"Go to Tasks\\\\\\">\\\\r\\\\n          <span class=\\\\\\"icon-list \\\\\\"></span>\\\\r\\\\n        </button>\\\\r\\\\n      </li>\\\\r\\\\n      <li class=\\\\\\"navigation__item\\\\\\">\\\\r\\\\n        <button class=\\\\\\"navigation__btn js-header-nav-btn\\\\\\" data-link=\\\\\\"reports/day/tasks\\\\\\" title=\\\\\\"Go to Reports\\\\\\">\\\\r\\\\n          <span class=\\\\\\"icon-statistics \\\\\\"></span>\\\\r\\\\n        </button>\\\\r\\\\n      </li>\\\\r\\\\n      <li class=\\\\\\"navigation__item\\\\\\">\\\\r\\\\n        <button class=\\\\\\"navigation__btn js-header-nav-btn\\\\\\" data-link=\\\\\\"settings/pomodoros\\\\\\" title=\\\\\\"Go to Settings\\\\\\">\\\\r\\\\n          <span class=\\\\\\"icon-settings \\\\\\"></span>\\\\r\\\\n        </button>\\\\r\\\\n      </li>\\\\r\\\\n    </ul>\\\\r\\\\n  </nav>\\\\r\\\\n</header><main id=\\\\\\"main\\\\\\" class=\\\\\\"main main--static\\\\\\">\\\\r\\\\n  <div class=\\\\\\"tabs js-nav\\\\\\">\\\\r\\\\n  \\\\r\\\\n    <div class=\\\\\\"tab tab--right tab--center js-tab-top\\\\\\">\\\\r\\\\n        <button class=\\\\\\"tab__btn js-tablink first-main-tab \\\\\\" data-reports-period=\\\\\\"day\\\\\\">day</button>\\\\r\\\\n        <button class=\\\\\\"tab__btn js-tablink second-main-tab \\\\\\" data-reports-period=\\\\\\"week\\\\\\">week</button>\\\\r\\\\n        <button class=\\\\\\"tab__btn js-tablink \\\\\\" data-reports-period=\\\\\\"month\\\\\\">month</button>\\\\r\\\\n    </div>\\\\r\\\\n  </div>\\\\r\\\\n  \\\\r\\\\n  <div class=\\\\\\"main__reports\\\\\\">\\\\r\\\\n    <a href=\\\\\\"/#/task-list\\\\\\" class=\\\\\\"main__reports__link main__reports__link--left icon-arrow-left\\\\\\" title=\\\\\\"Go to Global List\\\\\\"></a>\\\\r\\\\n    <div class=\\\\\\"main__reports__container js-reports-container\\\\\\" id=\\\\\\"reports-container\\\\\\">\\\\r\\\\n    </div>\\\\r\\\\n  </div>\\\\r\\\\n  <div class=\\\\\\"tabs js-nav\\\\\\">\\\\r\\\\n  \\\\r\\\\n    <div class=\\\\\\"tab tab--right tab--center tab--bottom js-tab-bottom\\\\\\">\\\\r\\\\n        <button class=\\\\\\"tab__btn js-tablink first-main-tab \\\\\\" data-reports-category=\\\\\\"pomodoros\\\\\\">pomodoros</button>\\\\r\\\\n        <button class=\\\\\\"tab__btn js-tablink second-main-tab \\\\\\" data-reports-category=\\\\\\"tasks\\\\\\">tasks</button>\\\\r\\\\n    </div>\\\\r\\\\n  </div>\\\\r\\\\n</main>\\""`;
